ledger/ledger-consensual/src/main/kotlin/net/corda/v5/ledger/consensual/transaction/ConsensualTransactionBuilder.kt

```diff
@@ -0,0 +1,54 @@
+package net.corda.v5.ledger.consensual.transaction
+
+import net.corda.v5.base.annotations.DoNotImplement
+import net.corda.v5.ledger.consensual.ConsensualState
+import java.security.PublicKey
+import java.time.Instant
+
+/**
+ * Defines a builder for [ConsensualSignedTransaction]s.
+ *
+ * //TODO(below docs are from obsolete part, review them.)
+ * It is a transaction class that's mutable (unlike the others which are all immutable). It is
+ * intended to be passed around parties that may edit it by adding new states. Then, once the states
+ * are right, this class can be used as a holding bucket to gather signatures from multiple parties.
+ */
+@DoNotImplement
+interface ConsensualTransactionBuilder {
+    /**
+     * @property timestamp The timestamp of the [ConsensualTransactionBuilder].
+     */
+    val timestamp: Instant?
```

Comment:
> Should this be nullable?

It is null only until it gets filled populated with the next function

> What does this instant represent? 

It will be the transaction's timestamp.

Reply:
I have removed timestamp and withTimestamp.
They will be set in `signInitial()` calling time.